tests/route/test_route_flap.py

```diff
@@ -181,31 +197,53 @@ def test_route_flap(duthost, tbinfo, ptfhost, ptfadapter,
                         break
         pytest_assert(vlan_mac, 'dual-tor without vlan mac !!!')
     else:
-       vlan_mac = dut_mac
+        vlan_mac = dut_mac
 
-    #get dst_prefix_list and aspath
+    # get dst_prefix_list and aspath
     routes = namedtuple('routes', ['route', 'aspath'])
-    iproute_info = get_ip_route_info(duthost)
+    iproute_info = get_ip_route_info(asichost)
     dst_prefix_list = []
     route_prefix_len = get_route_prefix_len(tbinfo, common_config)
+    
     for route_prefix in iproute_info:
         if "/{}".format(route_prefix_len) in route_prefix:
-            # Use only multipath routes, othervise there will be announced new routes to T0 neigbours on t1 topo
-            multipath = iproute_info[route_prefix][0].get('multipath', False)
-            if multipath:
-                out = iproute_info[route_prefix][0].get('path').split(' ')
-                aspath = out[1:]
-                entry = routes(route_prefix, ' '.join(aspath))
-                dst_prefix_list.append(entry)
-
+            # multia-asics can have more than 1 routes in iproute_info[route_prefix], even single-asics have only 1
+            for route_per_prefix in iproute_info[route_prefix]:
+                # Use only multipath routes, othervise there will be announced new routes to T0 neigbours on t1 topo
+                multipath = route_per_prefix.get('multipath', False)
+                if multipath:
+                    out = route_per_prefix.get('path').split(' ')
+                    aspath = out[1:]
+                    entry = routes(route_prefix, ' '.join(aspath))
+                    dst_prefix_list.append(entry)
     route_to_ping = dst_prefix_list[0].route
-    dev = json.loads(duthost.shell('vtysh -c "show ip route {} json"'.format(route_to_ping))['stdout'])
-    dev_port = dev[route_to_ping][0]['nexthops'][0]['interfaceName']
-
+    cmd = ' -c "show ip route {} json"'.format(route_to_ping)
+    dev_port = None
+    if duthost.is_multi_asic:
+        for asic in duthost.frontend_asics:
+            dev = json.loads(asic.run_vtysh(cmd)['stdout'])
+            nexthops = dev[route_to_ping][0]['nexthops']
+            internal_peers = duthost.get_internal_bgp_peers()
+            for hop in nexthops:
+                is_internal_hop = hop['ip'] in internal_peers.keys()
+                if 'interfaceName' in hop.keys() and 'IB' not in hop['interfaceName'] and not is_internal_hop:
```

Comment:
Is the `and 'IB'` check ncessary ? Would it not be enough to just check for membership in internal_peers? 

The reason I ask is that if for any reason the naming scheme changes down the road, then this test will unnecessarily break. It is better not to depend on naming format.

Reply:
Okay thats fine with me.  Feel free to merge this if there are no otehr issues.

Reply:
this testcase seems flaky on arista testbed, depending on which ports/routes it takes

I'll check the issue described in https://github.com/sonic-net/sonic-mgmt/issues/7507 before merging

Reply:
this comment is addressed